tests/output.rs

```diff
@@ -67,9 +66,35 @@ fn strip_indent<'line>(line: &'line str, indent: &str) -> &'line str {
     &line[indent.len()..]
 }
 
+#[cfg(target_os = "macos")]
+const DYNAMIC_LIBRARY_EXTENSION: &str = ".dylib";
+#[cfg(all(unix, not(target_os = "macos")))]
+const DYNAMIC_LIBRARY_EXTENSION: &str = ".so";
+#[cfg(windows)]
+const DYNAMIC_LIBRARY_EXTENSION: &str = ".dll";
+
+#[cfg(unix)]
+fn get_dynamic_library_name(library_name: &str) -> String {
+    format!("lib{library_name}{}", DYNAMIC_LIBRARY_EXTENSION)
+}
+#[cfg(windows)]
+fn get_dynamic_library_name(library_name: &str) -> String {
+    format!("{library_name}{}", DYNAMIC_LIBRARY_EXTENSION)
+}
+
 fn parse_command_line(command_line: &str) -> Vec<String> {
     assert!(command_line.starts_with('$'));
-    shlex::split(&command_line[1..]).unwrap()
+    shlex::split(&command_line[1..])
+        .unwrap()
+        .iter()
+        .map(|arg| {
+            regex!(r#"lib(\S+)\.so$"#)
```

Comment:
This seems like it's relying on the `.so` extension? How does that work?

Reply:
Because we control the tests?

I guess how I'm thinking of it is that the tests are written "canonically" (for Linux?) and then we massage from there based on what actual platform it's running on

(I'm also guessing you just had it "backwards" and didn't really need that explanation)